python/CMakeLists.txt

```diff
@@ -336,6 +343,8 @@ if(PYARROW_BUILD_PARQUET_ENCRYPTION)
   else()
     message(FATAL_ERROR "You must build Arrow C++ with PARQUET_REQUIRE_ENCRYPTION=ON")
   endif()
+else()
+  set(CYTHON_COMPILE_TIME_ENV "PARQUET_ENCRYPTION_ENABLED=0")
 endif()
```

Comment:
How about using the existing `CYTHON_FLAGS` variable instead of adding a new `CYTHON_COMPILE_TIME_ENV` variable?

Reply:
Could you try `list(APPEND CYTHON_FLAGS "-E" "PARQUET_ENCRYPTION_ENABLED=0")`?

Reply:
Great!